### PR TITLE
feature/22621_generate_tokens

### PR DESCRIPTION
**Pull request by glickel**
Add generateResetPasswordToken

### CODE REVIEW

File: src/Admin/UserAdmin.php [hunk 1]
```diff
@@ -111,6 +115,36 @@ public function retrieve(string $user)
...
+     * @param string $user
+     * @return string
+     *
+     * @throws UserAdminException
+     * @throws ApiClientException
+     */
+    public function generateResetPasswordToken(string $user): string
+    {
+        $request = (new RequestDescriptor())
+            ->setUrl($this->buildUrl(self::API_USERS_PATH_INFO . "/" . $user . "/password/reset-token"))
```
Comment (fanshan) [inline src/Admin/UserAdmin.php]: url_encode $user here

File: src/Admin/UserAdmin.php [hunk 2]
```diff
@@ -111,6 +115,36 @@ public function retrieve(string $user)
...
+    /**
+     * Generate a reset password token by user email or username
+     *
+     * @param string $user
+     * @return string
+     *
+     * @throws UserAdminException
+     * @throws ApiClientException
+     */
+    public function generateResetPasswordToken(string $user): string
```
Comment (fanshan) [inline src/Admin/UserAdmin.php]: We can inject User instance or string (email or username)

File: src/Admin/UserAdminInterface.php [hunk 3]
```diff
@@ -43,4 +44,12 @@ public function edit(User $formerUser, User $newNewUser);
      * @return User
      */
     public function retrieve(string $user);
+
+    /**
+     * Generate a reset password token by user email or username or user entity
+     *
+     * @param string $user
```
Comment (fanshan) [inline src/Admin/UserAdminInterface.php]: Please update the docblock here (this is the interfaice documentation)

File: specs/admin/user.yaml [hunk 4]
```diff
@@ -7,9 +7,23 @@ servers:
...
+                - name: validation-email
+                  in: query
+                  schema:
+                      type: integer
+                      enum: [0, 1]
+                      default: 1
+                  required: false
+                  description: >
+                      Define if a validation registration email must be send:
+                        * `0`: Email is not send
```
Comment (Naabi57) [inline specs/admin/user.yaml]:
```suggestion
                        * `0`: Email is not sent
```

File: specs/admin/user.yaml [hunk 5]
```diff
@@ -7,9 +7,23 @@ servers:
...
+            parameters:
+                - name: validation-email
+                  in: query
+                  schema:
+                      type: integer
+                      enum: [0, 1]
+                      default: 1
+                  required: false
+                  description: >
+                      Define if a validation registration email must be send:
```
Comment (Naabi57) [inline specs/admin/user.yaml]:
```suggestion
                      Define if a validation registration email must be sent:
```

File: specs/admin/user.yaml [hunk 6]
```diff
@@ -7,9 +7,23 @@ servers:
...
+                        * `0`: Email is not send
+                        * `1`: Email is send
```
Comment (Naabi57) [inline specs/admin/user.yaml]:
```suggestion
                        * `1`: Email is sent
```

File: specs/admin/user.yaml [hunk 7]
```diff
@@ -149,6 +152,91 @@ paths:
...
+        parameters:
+            - $ref: "#/components/parameters/userParam"
+        get:
+            summary: Generate a token for reseting password
+            operationId: generateResetPasswordToken
+            tags:
+                - User
+            responses:
+                200:
+                    description: A reset password token generated
```
Comment (Naabi57) [inline specs/admin/user.yaml]:
```suggestion
                    description: A reset password token has been generated
```

File: specs/admin/user.yaml [hunk 8]
```diff
@@ -149,6 +152,91 @@ paths:
...
+        parameters:
+            - in: query
+              name: token
+              required: true
+              description: The reset password token to validate
+              schema:
+                  type: string
+              example: myresetpasswordtoken
+        get:
+            summary: Validate a token for reseting password
```
Comment (Naabi57) [inline specs/admin/user.yaml]:
```suggestion
            summary: Validate a token for resetting password
```

File: specs/admin/user.yaml [hunk 9]
```diff
@@ -149,6 +152,91 @@ paths:
...
+            operationId: validateResetPasswordToken
+            tags:
+                - User
+            responses:
+                200:
+                    description: The User instace linked into reset password token.
```
Comment (Naabi57) [inline specs/admin/user.yaml]:
```suggestion
                    description: The User instance linked into reset password token.
```

File: src/Admin/UserAdmin.php [hunk 10]
```diff
@@ -38,17 +38,21 @@ class UserAdmin extends AbstractApiClient implements UserAdminInterface
...
      *
      * @return User
      *
-     * @throws UserAdminException
      * @throws ApiClientException
-     * @throws ValidatorException
+     * @throws UserAdminException
      */
-    public function create(User $user)
+    public function create(User $user, bool $sendValidationEmail = true)
```
Comment (Naabi57) [inline src/Admin/UserAdmin.php]:
```suggestion
    public function create(User $user, bool $sendValidationEmail = true): User
```

File: src/Admin/UserAdmin.php [hunk 11]
```diff
@@ -192,7 +256,7 @@ protected function parseSendException(Throwable $exception)
     /**
      * @return string
      */
-    public function getUsername(): string
+    public function getUsername()
```
Comment (Naabi57) [inline src/Admin/UserAdmin.php]:
```suggestion
    public function getUsername(): string
```

File: src/Admin/UserAdmin.php [hunk 12]
```diff
@@ -210,7 +274,7 @@ public function setUsername(string $username): UserAdmin
     /**
      * @return string
      */
-    public function getPassword(): string
+    public function getPassword()
```
Comment (Naabi57) [inline src/Admin/UserAdmin.php]:
```suggestion
    public function getPassword(): string
```

**Pull request merged**